scripts/patch.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Patch `gatsby-theme-mdx1` which resolves `@mdx-js/mdx@1` for now,
+ * it results incorrect pkg.version in header, and warning about importing `@reach/skip-nav/styles.css`.
+ */
+const fs = require('fs')
+
+const headerFilename = 'node_modules/gatsby-theme-mdx1/src/components/header.js'
+
+fs.writeFileSync(
+  headerFilename,
+  fs
+    .readFileSync(headerFilename, 'utf8')
+    .replace(
+      `import pkg from '@mdx-js/mdx/package.json'`,
+      `import pkg from '../../../../packages/mdx/package.json'`
+    )
+)
+
+const skipNavFilename =
+  'node_modules/gatsby-theme-mdx1/src/components/skip-nav.js'
+
+const skipNavStyles = 'import "@reach/skip-nav/styles.css"'
+
+const skipNavContent = fs.readFileSync(skipNavFilename, 'utf8')
+
+if (!skipNavContent.startsWith(skipNavStyles)) {
+  fs.writeFileSync(skipNavFilename, skipNavStyles + '\n' + skipNavContent)
+}
```

Comment:
could you expand on why this is needed?
It seems like versions are something which should be managed by package.json?
It also doesn't seem to be related to the PR's stated goal of frontmatters?

Reply:
I’ve commented at the front.

The document is generated by gastby, and it is not compatible with mdx@2 for now, so I hacked to use `gastby-theme-mdx1@npm:gastby-theme-mdx@^1.0.0` to build docs successfully.

And it requires `@mdxjs/react@1`, which results current version in title is always 1.x even it’s for 2.0.

So I have to hack like this to generate document correctly.

Reply:
Could this be solved with yarn resolutions?
Would it make sense to have this in its own PR?
It feels like this PR now captures three distinct change sets, fixing the frontmatter docs, adjusting the eslint settings, and updates to the documentation site.
All are good and welcome.

Reviewing them all together, especially when the title and description has no mention of the latter to is a bit confusing

Reply:
The goal of this PR is to generate correct document for `mdx@2`.

You can check https://mdxjs.com/ and https://mdx-git-docs-frontmatter-mdx.vercel.app/ to see the difference on page header.

I’m on iPad, so I can capture screenshot for you.

Reply:
> Could this be solved with yarn resolutions?

No, we are developing `gastby-theme-mdx@2` in this repo at the same time. And I don’t know how `resolutions` be related to this patch which changes specific files in node_modules.

Reply:
> So I have to hack like this to generate document correctly

Would something like https://github.com/ds300/patch-package work here?

Reply:
I’ve changed the title and description to clarify.

Reply:
> Would something like https://github.com/ds300/patch-package work here?

Maybe, but I’d prefer patch as-is for such simple task. (less dev pkg is always better)

Reply:
Not necessarily, those packages are included in patch-package because it can detect and warn if the package it is patching has changed, and warn the user.
The patch script as is, would include the changes regardless of it they fit/work with the version of the package they are patching, with no warning of possible conficts.

Reply:
`gastby-theme-mdx@1` will never change anymore. We’ve stopped developing on v1.

Reply:
@ChristianMurphy Any further discussion?

Reply:
I still think that using patch-packages is a clearer approach.
I'm cautious about a hand coded solution.

Reply:
An extra dev dependcy is not clearer to me, and I don't think it should block this PR.

Reply:
I understand your opinion and I respectfully disagree.
I see it as a well know problem with a well known solution (`patch-package`).
I don't see writing a custom solution as a worth while tradeoff to avoid some dev dependencies.

Reply:
Well, if you want `patch-package` strongly, can you raise another PR for it by yourself? It is good solution, but I don't want it for such simple task in this PR.

Reply:
Another mdx maintainer can approve and merge it if they want.
I'm not comfortable merging it as is.

Reply:
So why you don't want another PR by yourself? It's just a personal taste, I'm fine if you want it and implement it by yourself, I just don't want it in this PR.

Reply:
> I just don't want it in this PR.

Okay :man_shrugging: 
I'm still not comfortable approving and merging as is.
For the points outlined several times through this review; this PR has an overly broad scope that changes many unrelated pieces which could/should be discussed and changed independently, and it introduces some custom build scripts to patch packages.
That said, I'm also not blocking the PR either, if another maintainer feels strongly another way, they can approve and merge if they want.

> So why you don't want another PR by yourself?

If it gets merged as-is, I likely will.